### PR TITLE
Prepare 3.0 version

### PR DESCRIPTION
**Pull request by cristianoc72**
-  __PHP 8+ :__ large use of new PHP 8 features and functions
-   __Bump dependencies__
-  __Refactor tag constructors__ : refactor tag constructors to avoid [unsafe instantiations](https://psalm.dev/docs/running_psalm/issues/UnsafeInstantiation/). Now Psalm analysis is green at level 1 (the more strict level)
-   __Refactor TagFactory__ : now, to add a new tag it's enough to create the class and put it into `src/tags` directory
-  __Documentation Api__ : remove the script to create the documentation api locally, since `sami` does not support PHP 8. Add a Github Actions workflow to create and deploy it into the repository Github Pages (@gossi please, add a personal token to this repository secrets, called `DOCBLOCK_TOKEN`, to allow the deployment)
-  __Update tags docblock comments__ : update tags comments and link to the official references, to correctly create the documentation api

### CODE REVIEW

**Review by gossi — APPROVED** [review]
Looks really good. See my one comment if this is worth another change.

**Comment by gossi** [comment]
I added the token, hope that was the correct one.

**Comment by cristianoc72** [comment]
About the api generation workflow: imho it could be better to trigger it when publishing a new release, instead of when push on master branch. What do you think about?

**Comment by gossi** [comment]
yeah, makes sense.

File: src/Docblock.php [hunk 1]
```diff
@@ -328,9 +319,9 @@ public function hasTag(string $tagName): bool {
 	 *
 	 * @return ArrayList the tags
 	 */
-	public function getTags(string $tagName = null): ArrayList {
-		return $this->tags->filter(function (AbstractTag $tag) use ($tagName): bool {
-			return $tagName === null || $tag->getTagName() == $tagName;
+	public function getTags(string $tagName = ''): ArrayList {
+		return $tagName === '' ? $this->tags : $this->tags->filter(function (AbstractTag $tag) use ($tagName): bool {
```
[the remark this answers is not in the document]
Comment (cristianoc72) [reply src/Docblock.php]: Yep! Done.

**Pull request merged**